conversions/hexadecimal_to_octal2.c

```diff
@@ -0,0 +1,116 @@
+/**
+ * @file
+ * @brief Convert hexadecimal number to octal number
+ * @details
+ * The input is valid from 0 to 0xFFFF_FFFF_FFFF_FFFF.
+ *
+ * At first, this program converts a hex string to an unsigned long long
+ * decimal, and then to an octal string.
+ *
+ * When there is an invalid character in input string, this program stops
+ * parsing and converts the string until that character.
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+
+/**
+ * Convert a hexadecimal number to octal number.
+ * @param hex Hexadecimal number to convert.
+ * @returns A pointer to the converted octal string.
+ */
+char *hex_to_oct(const char *hex)
+{
+#define MAX_OCT_STR_LEN 23 /* 17_7777_7777_7777_7777_7777 */
+    char *octal = (char *)malloc(sizeof(char) * MAX_OCT_STR_LEN);
```

Comment:
since you are returning the pointer, there is no guarantee that the dynamically allocated memory would be available. Better to make it static and since the length is already known, this can be simplified.
```suggestion
    char octal[MAX_OCT_STR_LEN];
    memset(octal, '\0', MAX_OCTAL_STR_LEN); // initialize as NULL string
```

Reply:
Wait, isn't this returning an address of a local variable? The content of memory pointed by `octal` can be modified after returning from the function.

Reply:
you are correct. As I mentioned in the comment description, the variable must be defined as `static`. I forgot to make that chance in the suggestion code (rectified).